tests/test_pytypes.py

```diff
@@ -955,3 +955,21 @@ def test_fn_annotations(doc):
         doc(m.annotate_fn)
         == "annotate_fn(arg0: Callable[[list[str], str], int]) -> None"
     )
+
+
+def test_generics_compatibility(doc):
```

Comment:
Suppressing exceptions (`AttributeError` here) can be a big trap when refactoring.

Safer and more outstanding:

```
@pytest.mark.skipif(
    not m. if_defined__cpp_nontype_template_parameter_class, reason="C++20 feature not available."
)
```

I'd probably combine all tests into one `def test_typevar()` to not have repeat `skipif`. (Generally granularity is better, but in this case I'd come down on the side of making the test code easier to read and maintain.)